experiment/simulation/js/main.js

```diff
@@ -1 +1,411 @@
+'use strict';
+let message_flag = 0; // controls the changing of the message
+const messages = ["  Click on the beaker to Pour Water.  ",
+    "  Click on the FeCl3 powder to take a spoon of salt.  ",
+    "  Click on the spoon to add salt to water.  ",
+    "  Click on the Glass Rod to stir the FeCl3 solution.  ",
+    "  Click on the Conical flask containing Water to place it on the Magnetic stirrer/heater.  ",
+    "  Click on the Magnetic bead to drop it into the Flask.  ",
+    "  Click on the Heater knob to switch on the Heater.  ",
+    "  Click on the Stirrer knob to switch on the Magnetic Stirrer.  ",
+    "  Wait till Water starts boiling...  ",
+    "  Click on the Heater knob to switch off the Heater.  ",
+    "  Click on the Pipette to take 10ml of conc. FeCl3 solution.  ",
+    "  Click on the Pipette again to take 10ml of conc. FeCl3 solution.  ",
+    "  Click on the Pipette to add the FeCl3 to the hot water in conical flask.  ",
+    "  Click on the Pipette again to add the FeCl3 drop by drop into the conical flask.  ",
+    "  Fe(OH)3 colloid is obtained...  "
+];
+
+let message = messages[0];
+document.getElementById("message").innerHTML = message;
+let knob_heater_flag = 0; // turn-on or turn-off the heater
+let pipette_flag = 0; // handles the movement of pipette
+// pipette_flag guide:
+// 0 - no animation of pipette
+// 1 - filling of pipette with solution
+// 2 - motion of pipette to flask
+// 3 - dropping solution from pipette into flask
+// 4 - signifies end state
+
+function pour() {
+    const image = document.getElementById("top-beaker");
+    image.style.pointerEvents = 'none';
+    const animation_1 = anime.timeline({
+        targets: "#top-beaker",
+        duration: 800,
+        easing: 'linear'
+    });
+    animation_1.add({
+        translateX: '8vw',
+        translateY: '5vw'
+    }).add({
+        delay: '800',
+        rotateZ: '-90'
+    }).add({
+        update: function(anim) {
+            message = messages[1];
+            document.getElementById("message").innerHTML = message;
+        }
+    }).add({
+        delay: '500',
+        opacity: '0',
+    });
+
+    const animation_2 = anime.timeline({
+        targets: "#water",
+        duration: 800,
+        easing: 'linear'
+    });
+    animation_2.add({
+        translateX: '8vw',
+        translateY: '5vw'
+    }).add({
+        delay: '800',
+        rotateZ: '-90',
+        opacity: '0',
+    });
+
+    const animation_3 = anime.timeline({
+        targets: "#water-main-beaker",
+        duration: 800,
+        easing: 'linear',
+    }).add({
+        delay: '1800',
+        opacity: '0.4',
+    });
+
+    const animation_4 = anime.timeline({
+        targets: "#salt",
+        duration: 500,
+    }).add({
+        delay: '2500',
+        opacity: '1',
+    });
+
+    const animation_5 = anime.timeline({
+        targets: "#flask",
+        duration: 400,
+        easing: 'linear',
+    }).add({
+        opacity: 0.4,
```

Comment:
opacity is defined as a string everywhere else, if both ways are correct, choose one for consistency

Reply:
Changed datatype to string everywhere for consistency